### ISSUE TITLE
Run startSession steps in parallel

### ISSUE THREAD

**Issue by obeletski** [w3c/presentation-api#148]
startSession algorithm mentions two distinctive steps:
- Monitor the list of available presentation displays and wait for completion
-  Select the display and request permission after all available devices are found

Better user experience would be to have those two tasks executed in parallel and dynamically update the selection list while devices are being discovered.

WDYT, does it make sense to introduce that kind of change in the spec?

**Comment by mfoltzgoogle** [comment]
It does make sense to allow the user agent to start the "request permission" task in parallel with the "request compatible presentation displays" tasks.  I would suggest looking at some other specs to see how to describe parallel task execution in prose.  The revised algorithm should handle the following:

- Updating the selection as more displays are found
- Handles termination if no displays are found
- Handles termination of both tasks if the user cancels
 
Feel free to send a PR - I am a bit short of time to compose PRs for the time being.

**Comment by mfoltzgoogle** [comment]
Doing this in parallel makes sense.  It's not clear that the "monitoring displays" algorithm has a clear termination condition.  I could be roaming among WiFi networks with the dialog open and come across a display, or some other discovery process may complete in the background.

If change to parallel discovery and selection, then the page won't get a rejected Promise from `start()` until the user sees there are no displays (or gives up waiting) and explicitly cancels selection.  That seems okay to me.

So I would propose the following:

1. Always start monitoring for displays when `start()` is called
2. In parallel, notify the user with any displays found (or no displays), and allow the user to cancel
3. Cancelling will stop monitoring and reject the `Promise`.

I'll do this in a separate PR since it involves some surgery to the start presentation algorithm.

**Comment by tidoust** [comment]
@mfoltzgoogle, @obeletski,

I believe this issue has been addressed and may be closed. The `start()` procedure is now clear that the monitoring of displays and the selection of a display happen in parallel.

Please re-open the issue if you believe further changes are required.
